ironmon_tracker/data/TrainerData.lua

```diff
@@ -75,17 +124,329 @@ function TrainerData.initialize()
+			[305] = true,
+			[306] = true,
+			[308] = true,
+			[309] = true,
+			[310] = true,
+			[311] = true,
+			[314] = true,
+			[315] = true,
+			[316] = true,
+			[317] = true,
+			[328] = true,
+			[329] = true,
+			[330] = true,
+			[331] = true,
+			[341] = true,
+			[346] = true,
+			[347] = true,
+			[348] = true,
+			[349] = true,
+			[354] = true,
+			[355] = true,
+			[356] = true,
+			[357] = true,
+			[360] = true,
+			[361] = true,
+			[362] = true,
+			[363] = true,
+			[365] = true,
+			[366] = true,
+			[367] = true,
+			[368] = true,
+			[370] = true,
+			[371] = true,
+			[372] = true,
+			[373] = true,
+			[379] = true,
+			[380] = true,
+			[381] = true,
+			[382] = true,
+			[388] = true,
+			[389] = true,
+			[390] = true,
+			[391] = true,
+			[393] = true,
+			[394] = true,
+			[395] = true,
+			[396] = true,
+			[409] = true,
+			[410] = true,
+			[411] = true,
+			[412] = true,
+			[421] = true,
+			[422] = true,
+			[423] = true,
+			[424] = true,
+			[430] = true,
+			[431] = true,
+			[432] = true,
+			[433] = true,
+			[437] = true,
+			[438] = true,
+			[439] = true,
+			[440] = true,
+			[456] = true,
+			[462] = true,
+			[466] = true,
+			[467] = true,
+			[468] = true,
+			[477] = true,
+			[478] = true,
+			[479] = true,
+			[480] = true,
+			[482] = true,
+			[485] = true,
+			[486] = true,
+			[487] = true,
+			[488] = true,
+			[489] = true,
+			[497] = true,
+			[498] = true,
+			[499] = true,
+			[500] = true,
+			[515] = true,
+			[516] = true,
+			[517] = true,
+			[518] = true,
+			[541] = true,
+			[542] = true,
+			[543] = true,
+			[544] = true,
+			[548] = true,
+			[549] = true,
+			[550] = true,
+			[551] = true,
+			[555] = true,
+			[556] = true,
+			[557] = true,
+			[558] = true,
+			[562] = true,
+			[563] = true,
+			[564] = true,
+			[565] = true,
+			[607] = true,
+			[608] = true,
+			[609] = true,
+			[610] = true,
+			[622] = true,
+			[623] = true,
+			[624] = true,
+			[625] = true,
+			[633] = true,
+			[634] = true,
+			[636] = true,
+			[637] = true,
+			[638] = true,
+			[639] = true,
+			[643] = true,
+			[644] = true,
+			[645] = true,
+			[646] = true,
+			[657] = true,
+			[658] = true,
+			[659] = true,
+			[660] = true,
+			[682] = true,
+			[683] = true,
+			[684] = true,
+			[685] = true,
+			[688] = true,
+			[689] = true,
+			[690] = true,
+			[691] = true,
 		}
+		for i=770, 801, 1 do
+			trainerIds[i] = true
+		end
+		for i=805, 847, 1 do
+			trainerIds[i] = true
+		end
+		trainerIds[851] = true
+		trainerIds[852] = true
+		trainerIds[853] = true
+		trainerIds[854] = true
+		trainerIds[855] = true
+		return trainerIds
+	elseif GameSettings.game == 3 then -- FireRed/LeafGreen
+		local trainerIds = {}
+		for i=1, 88, 1 do
+			trainerIds[i] = true
+		end
+		for i=492, 497, 1 do
+			trainerIds[i] = true
+		end
+		for i=511, 515, 1 do
+			trainerIds[i] = true
+		end
+		trainerIds[147] = true
+		trainerIds[200] = true
+		trainerIds[263] = true
+		trainerIds[530] = true
```

Comment:
Perhaps a cleaner / more compact way of defining these could be to define the ranges of dummy trainer ids and then looping through to define the `[id] = true` array items, e.g. something like:
```lua
local idRanges = {
    { 40, 43 },
    { 47, 50 }.
    ...
    { 851, 855 },
}
for _, range in pairs(idRanges) do
    for i=range[1], range[2], 1 do
        trainerIds[i] = true
    end
end
```
of course this table of ranges could be written out however you feel looks best, and any specific single IDs that don't lie within a range could be added after this loop *or* in the `idRanges` as `{ id, id }`

Reply:
This is a good suggestion. I ended up doing just this in one of the later additions to a long list of items, as seen in the Ruby/Sapphire trainer mappings in the same file.

I'll get this included.

Reply:
Resolved https://github.com/besteon/Ironmon-Tracker/pull/403/commits/da72c7752034b834bf6730af2b2b11957766894c